completions/fish/wl-paste.fish

```diff
@@ -0,0 +1,50 @@
+function __wayland_seats --description 'Print all wayland seats'
+    if type -q weston-info
+        weston-info 2>/dev/null | sed -n '/wl_seat/{n;s/\s*name: //;p}'
+    else if type -q loginctl
+        loginctl --no-legend --no-pager list-seats 2>/dev/null
+    else
+        # Fallback seat
+        echo "seat0"    
+    end
+end
+
+function __wl_paste_types --description 'Print types with context'
+    set -l type (commandline -o)
+
+    for i in (seq (count $type))
+        if [ "$type[$i]" = '--primary' ]; or [ "$type[$i]" = '-p' ]
+            set clip "primary"
+        else if [ "$type[$i]" = '--seat' ]; or [ "$type[$i]" = '-s' ]
+            set seat $type[(math $i + 1)]
+        end
+    end
+
+    # Note fish does not handle passing unset variables
+    # to commands well, thus setting clip to "--primary" and passing
+    # that to wl-paste wont work, so if statements are used instead
+    if test -n "$seat"; and test -n "$clip"
```

Comment:
Probably need to add a check to see if clipboard is empty so `No Selection` isn't offered as a type.

Reply:
Nevermind, `No selection` is sent to `stderr` so it can just be redirected to NULL

Reply:
Of course. It's an error, not the output.

Still, if it does end up outputting an error for one reason or another, maybe we should swallow it (`2>/dev/null`) instead of printing it to shell's real stderr.

Reply:
Yeah that's how I eventually ended up handling it.